climada/hazard/centroids/centr.py

```diff
@@ -1055,151 +504,258 @@ def select_mask(self, reg_id=None, extent=None):
 
         Parameters
         ----------
-        file_data : str or h5
-            If string, path to write data. If h5 object, the datasets will be generated there.
+        file_name : str
+            vector file with format supported by fiona and 'geometry' field.
+        dst_crs : crs, optional
+            reproject to given crs
+            If not crs is given in the file, simply sets the crs.
+
+        Returns
+        -------
+        centr : Centroids
+            Centroids with points according to the given vector file
         """
-        if isinstance(file_data, str):
-            LOGGER.info('Writing %s', file_data)
-            with h5py.File(file_data, 'w') as data:
-                self._write_hdf5(data)
-        else:
-            self._write_hdf5(file_data)
-
-    def _write_hdf5(self, data):
-        str_dt = h5py.special_dtype(vlen=str)
-        for centr_name, centr_val in self.__dict__.items():
-            if isinstance(centr_val, np.ndarray):
-                data.create_dataset(centr_name, data=centr_val, compression="gzip")
-            elif centr_name == 'meta' and centr_val:
-                centr_meta = data.create_group(centr_name)
-                for key, value in centr_val.items():
-                    if value is None:
-                        LOGGER.info("Skip writing Centroids.meta['%s'] for it is None.", key)
-                    elif key not in ('crs', 'transform'):
-                        if not isinstance(value, str):
-                            centr_meta.create_dataset(key, (1,), data=value, dtype=type(value))
-                        else:
-                            hf_str = centr_meta.create_dataset(key, (1,), dtype=str_dt)
-                            hf_str[0] = value
-                    elif key == 'transform':
-                        centr_meta.create_dataset(
-                            key, (6,),
-                            data=[value.a, value.b, value.c, value.d, value.e, value.f],
-                            dtype=float)
-            elif centr_name == 'geometry':
-                LOGGER.debug("Skip writing Centroids.geometry")
+
+        centroids = cls.from_geodataframe(gpd.read_file(file_name))
+        if dst_crs is not None:
+            if centroids.crs:
+                centroids.gdf.to_crs(dst_crs, inplace=True)
             else:
-                LOGGER.info("Skip writing Centroids.%s:%s, it's neither an array nor a non-empty"
-                            " meta object", centr_name, centr_val.__class__.__name__)
-        hf_str = data.create_dataset('crs', (1,), dtype=str_dt)
-        hf_str[0] = CRS.from_user_input(self.crs).to_wkt()
+                centroids.gdf.set_crs(dst_crs, inplace=True)
+        return centroids
+
+    #TODO: Check whether other variables are necessary, e.g. dist to coast
+    @classmethod
+    def from_csv(cls, file_path, crs=DEF_CRS, var_names=None):
+        """
```

Comment:
But where do we store the defaults? This is potentially dangerous if we add/change one of these names in the future since there are more than one method that use them, so they can be missed easily.

Reply:
@chahank I think this is an outdated comment, the csv and excel methods work by using **kwargs and the private method _from_dataframe, which you wrote

Reply:
aaaa excellent!